rust/protocol/src/sealed_sender.rs

```diff
@@ -707,12 +707,11 @@ mod sealed_sender_v1 {
         )?;
 
         // Encrypt the sender's public key with AES-256 CTR and a MAC.
-        let sender_static_key_ctext = crypto::aes256_ctr_hmacsha256_encrypt(
+        let sender_static_key_ctext = crypto::ctr::aes_256_ctr_hmac_encrypt(
             &sender_identity.public_key().serialize(),
             &sender_eph_keys.cipher_key,
-            &sender_eph_keys.mac_key,
-        )
-        .expect("just generated these keys, they should be correct");
+            array_ref![&sender_eph_keys.mac_key, 0, crypto::ctr::MAC_KEY_LENGTH],
```

Comment:
…and this is where the MAC key is incorrectly truncated, and our lack of "decrypting this existing message with this session state" tests has come back to bite us.